c++/src/kj/test.h

```diff
@@ -106,7 +106,7 @@ class TestCase {
   do { \
     KJ_IF_MAYBE(e, ::kj::runCatchingExceptions([&]() { code; })) { \
       KJ_EXPECT(::kj::_::hasSubstring(e->getDescription(), message), \
-          "exception description didn't contain expected substring", *e); \
+          "exception description didn't contain expected substring", *e, message); \
```

Comment:
Not to mention, because `e` is an exception, I think the whole stack trace will be rendered before `; the desired substring` is appended to the end?

Reply:
> It's redundant, since the log message will already contain the text hasSubstring(e->getDescription(), "the desired substring")

It's not; `message` is the actual substring, not the expected substring. For example:

```
It's redundant, since the log message will already contain the text hasSubstring(e->getDescription(), "the desired substring")

I'll revert it, but would note that it successfully surfaced the information we need.

Reply:
> It's redundant, since the log message will already contain the text hasSubstring(e->getDescription(), "the desired substring")

I don't see it anywhere in the message that tls-test printed out:
```
2: kj\compat\tls-test.c++:705: failed: expected ::kj::_::hasSubstring(e->getDescription(), message); exception description didn't contain expected substring; *e = kj\compat\tls.c++:215: failed: TLS peer's certificate is not trusted; reason = Hostname mismatch
```

I see `reason`, but that's part of the exception.

It does in fact append the expected substring after the whole stack. I'll revert it, but would note that it seems to have surfaced the information we need for debugging.

Reply:
Reverted. Once the non-Windows builds are green I'll merge.

Reply:
Whoa weird, since `message` is a macro parameter I expected it to be substituted with its value, but indeed it appears this is not the case... sorry, maybe this was a good change after all.

Reply:
Oh wait, I see now, in `tls-test.c++` there is this code:

```
KJ_EXPECT_THROW_MESSAGE(message, clientPromise.wait(test.io.waitScope));
```

So the macro parameter `message` is actually expanding to the variable name `message` in this one case. So the right fix is to edit this call site, I think.